main.go

```diff
@@ -64,8 +63,8 @@ func main() {
 	exitStatus, err := c.Run()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error executing CLI: %s\n", err.Error())
-		os.Exit(1)
```

Comment:
Note that previously, the exit code of AgentCommand.Run() was not being returned after an error.  The returned exit code was either 0 or 1.  My current changes have the system returning the exit code generated by AgentCommand.Run().  I'm not sure if this is the behavior you want